presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestMergeWindows.java

```diff
@@ -464,19 +468,21 @@ public void testNotMergeDifferentOrdering()
                                         .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(QUANTITY_ALIAS))),
                                 window(windowMatcherBuilder -> windowMatcherBuilder
                                                 .specification(specificationA)
-                                                .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(EXTENDEDPRICE_ALIAS)))
-                                                .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(DISCOUNT_ALIAS))),
+                                                .addFunction(functionCall("sum", UNSPECIFIED_FRAME, ImmutableList.of(EXTENDEDPRICE_ALIAS)))
```

Comment:
> So are you objecting to the fact that WindowNode.Function.FunctionCall.Window.getFrame() returns empty frame?

No. I'm objecting to the fact that prior to this change we were validating that the plan contained the frame information given in the query, and now we are not.

Because the object returned by `WindowNode.Function.FunctionCall.Window.getFrame()` is a parser object, it may be the case that validating the plan by checking the frame information there was never really correct (If so, I apologize; I'm reasonably certain I wrote that validation code originally), but that it worked as a side-effect of how the parser was implemented.

This branch changes the parser in a way that we are fundamentally in agreement is (probably) correct. As a side-effect of the changes, validating the frame information in the plan by looking at the result of calling `WindowNode.Function.FunctionCall.Window.getFrame()` may no longer be the correct approach (if it ever was in the first place).

As part of this change, we need to update the plan matching framework so that we can continue to validate that frame information in the SQL query is propagated into the plan somewhere, possibly in `WindowNode.Function.Frame`. If the frame information in the query is in there, it's a matter of updating the framework to look there instead of in the parser object. If not, there's a bug in the code that needs to be resolved.

Before this can be merged, we need to validate that frame information from the SQL query is making it into the plan. As the code stands, we are not doing that.